Lib/test/test_os.py

```diff
@@ -735,6 +737,85 @@ def test_stat_result_pickle(self):
+
+        time_attributes = 'st_atime st_mtime st_ctime st_birthtime'.split()
+        self.check_timestamp_agreement(result, time_attributes)
+
+        # Check that valid attributes match os.stat.
+        requirements = (
+            ('st_mode', os.STATX_TYPE | os.STATX_MODE),
+            ('st_nlink', os.STATX_NLINK),
+            ('st_uid', os.STATX_UID),
+            ('st_gid', os.STATX_GID),
+            ('st_atime', os.STATX_ATIME),
+            ('st_atime_ns', os.STATX_ATIME),
+            ('st_mtime', os.STATX_MTIME),
+            ('st_mtime_ns', os.STATX_MTIME),
+            ('st_ctime', os.STATX_CTIME),
+            ('st_ctime_ns', os.STATX_CTIME),
+            ('st_ino', os.STATX_INO),
+            ('st_size', os.STATX_SIZE),
+            ('st_blocks', os.STATX_BLOCKS),
+            ('st_birthtime', os.STATX_BTIME),
+            ('st_birthtime_ns', os.STATX_BTIME),
+            # unconditionally valid members
+            ('st_blksize', 0),
+            ('st_dev', 0),
+            ('st_rdev', 0),
+        )
+        basic_result = os.stat(self.fname)
+        for name, bits in requirements:
+            if result.stx_mask & bits == bits:
+                x = getattr(result, name)
+                b = getattr(basic_result, name)
+                if isinstance(x, float):
+                    self.assertAlmostEqual(x, b, msg=name)
+                else:
+                    self.assertEqual(x, b, msg=name)
+
+        # Access all the attributes multiple times to test cache refcounting.
+        members = [name for name in dir(result)
+                   if name.startswith('st_') or name.startswith('stx_')]
+        for _ in range(10):
+            for name in members:
+                getattr(result, name)
+
+        for name in members:
+            try:
+                setattr(result, name, 1)
+                self.fail("No exception raised")
+            except AttributeError:
+                pass
+
+        self.assertEqual(result.stx_attributes & result.stx_attributes_mask,
+                         result.stx_attributes)
+
+    @unittest.skipUnless(hasattr(os, 'statx'), 'test needs os.statx()')
+    def test_statx_attributes(self):
+        self.check_statx_attributes(self.fname)
+
+    @unittest.skipUnless(hasattr(os, 'statx'), 'test needs os.statx()')
+    def test_statx_attributes_bytes(self):
+        try:
+            fname = self.fname.encode(sys.getfilesystemencoding())
+        except UnicodeEncodeError:
+            self.skipTest("cannot encode %a for the filesystem" % self.fname)
+        self.check_statx_attributes(fname)
+
+    @unittest.skipUnless(hasattr(os, 'statx'), 'test needs os.statx()')
+    def test_statx_sync(self):
+        # Test sync= kwarg parsing.  (We can't predict if or how the result
+        # will change.)
+        for sync in (False, True):
+            with self.subTest(sync=sync):
+                os.statx(self.fname, os.STATX_BASIC_STATS, sync=sync)
+
```

Comment:
Will need more test cases for different ways of constructing/using statx (path_fd, dir_fd, PathLike)

Reply:
Should the dir_fd tests go in here in test_os.py or in TestPosixDirFd in test_posix.py? 

Should all the statx tests be in test_posix.py?  The comment at the top of test_os.py says it's for "a few functions which have been determined to be more portable than they had been thought to be", and statx is not.

Reply:
I think in `test_os` works; don't need to test that "do the dir_fd helpers use handle every case right", want to just make sure "if someone deleted the dir_fd handling code; a test would fail"

Reply:
`test_posix` seems to be around https://docs.python.org/3/library/posix.html; I don't have experience / knowledge around that so not sure